scripts/ScriptUtilities.psm1

```diff
@@ -33,19 +33,89 @@ function Exit-WithFailureMessage($scriptName, $message) {
     exit 1
 }
 
+# NuGet Package Creation section
+function New-NuGetPackageFromProjectFile($Configuration, $project, $version) {
+    $PackageOutputDirectory = "$BinRoot\NuGet\$Configuration"
+    $projectFile = "$SourceRoot\$project\$project.csproj"
+
+    $arguments =
+        "pack", $projectFile,
+        "--configuration", $Configuration,
+        "--no-build", "--no-restore",
+        "--include-source", "--include-symbols",
+        "-p:Platform=$Platform",
+        "--output", $PackageOutputDirectory
+
+    Write-Debug "dotnet $($arguments -join ' ')"
+
+    dotnet $arguments
+    if ($LASTEXITCODE -ne 0) {
+        Exit-WithFailureMessage $ScriptName "$project NuGet package creation failed."
+    }
+}
+
+function New-NuGetPackageFromNuspecFile($Configuration, $project, $version, $suffix = "") {
+    $PackageOutputDirectory = "$BinRoot\NuGet\$Configuration"
+    $nuspecFile = "$SourceRoot\NuGet\$project.nuspec"
+
+    $arguments=
+        "pack", $nuspecFile,
+        "-Symbols",
+        "-Properties", "configuration=$Configuration;version=$version",
+        "-Verbosity", "Quiet",
+        "-BasePath", ".\",
+        "-OutputDirectory", $PackageOutputDirectory
+
+    if ($suffix -ne "") {
+        $arguments += "-Suffix", $Suffix
+    }
+
+    $nugetExePath = "$RepoRoot\.nuget\NuGet.exe"
+
+    Write-Debug "$nuGetExePath $($arguments -join ' ')"
+
+    &$nuGetExePath $arguments
+    if ($LASTEXITCODE -ne 0) {
+        Exit-WithFailureMessage $ScriptName "$project NuGet package creation failed."
+    }
+
+    Write-Information "  Successfully created package '$BinRoot\NuGet\$Configuration\$Project.$version.nupkg'."
+}
+
+function New-NuGetPackages($Configuration, $Projects) {
+    $versionPrefix, $versionSuffix = & $PSScriptRoot\Get-VersionConstants.ps1
+    $version = "$versionPrefix-$versionSuffix"
+
+    # We can build the NuGet packages for library projects directly from their
+    # project file.
+    foreach ($project in $Projects.NewLibrary) {
+        New-NuGetPackageFromProjectFile $Configuration $project $version
+    }
+
+    # Unfortunately, application projects like MultiTool need to include things
+    # that are not specified in the project file, so their packages still require
+    # a .nuspec file.
+    foreach ($project in $Projects.NewApplication) {
+        New-NuGetPackageFromNuSpecFile $Configuration $project $version
+    }
+}
+
 $RepoRoot = $(Resolve-Path $PSScriptRoot\..).Path
+$Platform = "AnyCPU"
 $SourceRoot = "$RepoRoot\src"
 $NuGetPackageRoot = "$SourceRoot\packages"
 $JsonSchemaPath = "$SourceRoot\Sarif\Schemata\Sarif.schema.json"
 $BuildRoot = "$RepoRoot\bld"
 $BinRoot = "$BuildRoot\bin"
-
 $SarifExtension = ".sarif"
```

Comment:
>S [](start = 1, length = 1)

Ultra-nit: That blank line was there to separate the path variables from this variable that isn't a path.